contrib/pkg/output/snapshot.go

```diff
@@ -207,53 +208,91 @@ type Snapshot struct {
 	ListsToSync []ResourceList
 }
 
+// Options for writing resources of a given type
+type OutputOpts struct {
+	// error handler
+	ErrHandler ErrorHandler
+
+	// If provided, ensure the resource has been verified before adding it to snapshots
+	Verifier verifier.OutputResourceVerifier
+}
+
 // sync the output snapshot to local cluster storage.
 // only writes resources intended for the local cluster (with ClusterName == "")
 // Note that Syncing snapshots in this way adds the label
-func (s Snapshot) SyncLocalCluster(ctx context.Context, cli client.Client, errHandler ErrorHandler) {
+func (s Snapshot) SyncLocalCluster(ctx context.Context, cli client.Client, opts OutputOpts) {
 
+	var presentResources []ezkube.Object
 	for _, list := range s.ListsToSync {
 		listForLocalCluster := list.SplitByClusterName()[multicluster.LocalCluster]
 
+		for _, resource := range listForLocalCluster {
+			if opts.Verifier != nil {
+				gvk := resource.GetObjectKind().GroupVersionKind()
```

Comment:
we don't set these when we translate out objects, we may need to use `apiutil.GVKForObject` here. another option is to just pass the gvks in from generated code